CMakeLists.txt

```diff
@@ -67,7 +67,6 @@ set( with-modelset "full" CACHE STRING "The modelset to include. Sample configur
 set( with-models OFF CACHE STRING "The models to include as a semicolon-separated list of model headers (without the .h extension). This option is mutually exclusive with -Dwith-modelset. [default=OFF]." )
 set( tics_per_ms "1000.0" CACHE STRING "Specify elementary unit of time [default=1000 tics per ms]." )
 set( tics_per_step "100" CACHE STRING "Specify resolution [default=100 tics per step]." )
-set( external-modules OFF CACHE STRING "External NEST modules to be linked in, separated by ';', [default=OFF]." )
```

Comment:
Static linking is intentionally removed because it was complex to implement and as far as I know little used. We can re-instate it later if there should be actual demand.

Reply:
How do you come to the insight "as far as I know little used"? – Would it make sense to at least improve documentation on how to build modules with a notice that static modules are no longer supported and how users could change to dynamic modules?

Reply:
There are currently only 12 forks of the extension module (EM) repository, of which only four by people outside the core team, so the EM mechanism is little used outside of NESTML. We changed the interface with NEST 3.6 (new model registration scheme) and there has been no questions/comments/complaints about this since December. The static linking mechanism is mentioned in NEST only in the CMake flags documentation (I will check that I removed it there) and as final item in the Extension Module documentation (https://nest-extension-module.readthedocs.io/en/latest/extension_modules.html#linking-mymodule-into-nest). The process is rather cumbersome, so if it had been used, I am sure there would have been questions. But I did not find a single mail in the NEST User archives concerning the -Dexternal-modules switch.

Reply:
Just for posterity, what is the argument for removing this functionality? For example, could statically linked module names conflict with dynamically loaded names, as we check whether modules are already loaded on the basis of a list of dynamic module names which doesn't include the static ones?

Reply:
As per the 2024 Q1 NEST hackathon, we decided to remove support for statically linked extension modules for now. The management of potential naming conflicts, and adding the requisite support in combination with the new loading/unloading mechanism would be prohibitive. Furthermore, we are not aware of this mechanism being used in practice in the past (many) years. Please write to us on the NEST Simulator mailing list if you require this functionality and we can discuss it further.